src/bin/ion/commands/beta/generate/context.rs

```diff
@@ -20,14 +20,14 @@ impl CodeGenContext {
     }
 }
 
-/// Represents an abstract data type type that can be used to determine which templates can be used for code generation.
+/// A target-language-agnostic data type that determines which template(s) to use for code generation.
 #[derive(Debug, Clone, PartialEq, Serialize)]
 pub enum AbstractDataType {
-    // a struct with a scalar value (used for `type` constraint)
+    // A scalar value (e.g. a string or integer or user defined type)
```

Comment:
We can address this in another PR, but now that I know that this includes user defined types, it isn't really scalar. (For example: user type `struct Point2D {x: f64, y: f64}` holds two scalar values, so it is itself a composite value.)

It's not immediately clear to me why `Point2D` wouldn't be an `AbstractDataType::Struct`, but that's probably just my unfamiliarity.

Reply:
As per offline discussion scalar values are possible for user defined types as well. 
e.g. Given below ISL,
 ```IonSchema
 type::{
  name: value_type,
   type: int
 }
```
 Corresponding abstract type in Rust would look like following:
```rust
 struct ValueType {
    value: i64
}
```

Also added similar examples in the doc comments.